pandas/tests/test_multilevel.py

```diff
@@ -1932,6 +1932,69 @@ def test_repeat(self):
         m_df = Series(data, index=m_idx)
         assert m_df.repeat(3).shape == (3 * len(data),)
 
+    def test_subsets_multiindex_dtype(self):
+        # GH 20757
+        data = [
+            [
+                "n",
+                1,
+                0,
+                False,
+                2,
+                1,
+                False,
+                0,
+                0,
+                False,
```

Comment:
can you make this a bit more programatic, rather than hardcoding these values, it would  be easier to copy-paste

Reply:
Thanks for the feedback, @jreback! Reproduced the same behaviour with simpler input data.